ext/standard/proc_open.c

```diff
@@ -135,93 +135,24 @@ int openpty(int *master, int *slave, char *name, struct termios *termp, struct w
 
 #include "proc_open.h"
 
-static int le_proc_open; /* Resource number for `proc` resources */
+zend_class_entry *process_ce;
+static zend_object_handlers process_object_handlers;
 
-/* {{{ _php_array_to_envp
- * Process the `environment` argument to `proc_open`
- * Convert into data structures which can be passed to underlying OS APIs like `exec` on POSIX or
- * `CreateProcessW` on Win32 */
-static php_process_env _php_array_to_envp(zval *environment)
-{
-	zval *element;
-	php_process_env env;
-	zend_string *key, *str;
-#ifndef PHP_WIN32
-	char **ep;
-#endif
-	char *p;
-	size_t sizeenv = 0;
-	HashTable *env_hash; /* temporary PHP array used as helper */
-
-	memset(&env, 0, sizeof(env));
-
-	if (!environment) {
-		return env;
-	}
-
-	uint32_t cnt = zend_hash_num_elements(Z_ARRVAL_P(environment));
-
-	if (cnt < 1) {
-#ifndef PHP_WIN32
-		env.envarray = (char **) ecalloc(1, sizeof(char *));
-#endif
-		env.envp = (char *) ecalloc(4, 1);
-		return env;
-	}
-
-	ALLOC_HASHTABLE(env_hash);
-	zend_hash_init(env_hash, cnt, NULL, NULL, 0);
-
-	/* first, we have to get the size of all the elements in the hash */
-	ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(environment), key, element) {
-		str = zval_get_string(element);
-
-		if (ZSTR_LEN(str) == 0) {
-			zend_string_release_ex(str, 0);
-			continue;
-		}
-
-		sizeenv += ZSTR_LEN(str) + 1;
-
-		if (key && ZSTR_LEN(key)) {
-			sizeenv += ZSTR_LEN(key) + 1;
-			zend_hash_add_ptr(env_hash, key, str);
-		} else {
-			zend_hash_next_index_insert_ptr(env_hash, str);
-		}
-	} ZEND_HASH_FOREACH_END();
-
-#ifndef PHP_WIN32
-	ep = env.envarray = (char **) ecalloc(cnt + 1, sizeof(char *));
-#endif
-	p = env.envp = (char *) ecalloc(sizeenv + 4, 1);
-
-	ZEND_HASH_FOREACH_STR_KEY_PTR(env_hash, key, str) {
-#ifndef PHP_WIN32
-		*ep = p;
-		++ep;
-#endif
-
-		if (key) {
-			memcpy(p, ZSTR_VAL(key), ZSTR_LEN(key));
-			p += ZSTR_LEN(key);
-			*p++ = '=';
-		}
```

Comment:
I'm not sure why this chunk is changed or if it's changed at all and how it's related to the resurce->object change.
Can you please try to minimize the diff.

Reply:
Agree this just makes review more difficult

Reply:
you should just keep it in the same place and put the object related functions behind it.

Reply:
I think I managed to get rid of this piece of diff. :)